admiral/pkg/apis/admiral/model/dependencyproxy.proto

```diff
@@ -0,0 +1,53 @@
+syntax = "proto3";
+
+package admiral.global.v1alpha;
+
+option go_package = "model";
+
+// The below example of DependencyProxy
+//```yaml
+//    apiVersion: admiral.io/v1alpha1
+//    kind: DependencyProxy
+//    metadata:
+//      name: dependency-proxy-example
+//      namespace: admiral
+//      annotations:
+//          admiral.io/env: stage
+//    spec:
+//      destination:
+//          identity:
+//          dns_suffix: "xyz"
+//          dns_prefix:
+//              - "test0"
+//              - "test1"
+//      proxy:
+//          identity: nginx-gw
+//```
+//
+message DependencyProxy {
+    // Configuration of the destination identity for which the
+    // requests should be proxied.
+    Destination destination = 1;
+
+    // Configuration of the proxy's identity through which the requests
+    // to the destination will be proxied through.
+    Proxy proxy = 2;
+
+}
+
+message Destination {
+    // Identifier of the destination workload.
+    string identity = 1;
+
+    // An ordered list of all DNS prefixes.
+    repeated string dns_prefixes = 2;
+
+    // The DNS suffix that should be appended while
+    // constructing the endpoint of the destination service.
+    string dns_suffix = 3;
+}
+
+message Proxy {
+    // Identifier of the proxy's workload
+    string identity = 3;
```

Comment:
Just to confirm, admiral.io/env label on the dependency proxy would be used in addition to this identity to construct the proxy endpoint?

Reply:
yes, that is correct. Let me know if you want to make it more explicit to avoid confusion.

Reply:
I think we can add an example in the docs with a sample dependencyproxy and what Admiral renders